### PR TITLE
[NFC] cleanup

### PR DESCRIPTION
**Pull request by tomerd**
motivation: nicer code

changes:
* fix several typos
* removed some dead and deprecated code
* remove some TODO/FIXME that have been (or could be easily) addressed
* Foundation.URL -> URL (now that we deprecated TSC.URL)

### CODE REVIEW

**Comment by tomerd** [comment]
@swift-ci please smoke test

**Review by abertelrud — APPROVED** [review]
Nice!

File: Sources/PackageCollections/Storage/SQLitePackageCollectionsStorage.swift [hunk 1]
```diff
@@ -73,9 +74,8 @@ final class SQLitePackageCollectionsStorage: PackageCollectionsStorage, Closable
     }
 
     deinit {
-        // TODO: we could wrap the failure here with diagnostics if it wasn't optional throughout
         guard case .disconnected = (try? self.withStateLock { self.state }) else {
-            return assertionFailure("db should be closed")
+            return self.observabilityScope.emit(error: "db should be closed")
```
Comment (tomerd) [inline Sources/PackageCollections/Storage/SQLitePackageCollectionsStorage.swift:78]: @yim-lee are you okay with this change or prefer we continue to assert?

Comment (tomerd) [reply Sources/PackageCollections/Storage/SQLitePackageCollectionsStorage.swift:78]: (could probably use a better error message)

**Comment by tomerd** [comment]
@swift-ci please smoke test

**Comment by tomerd** [comment]
@swift-ci please smoke test

File: Sources/PackageCollections/Storage/SQLitePackageCollectionsStorage.swift (hunk 1 above)
Comment (yim-lee) [reply Sources/PackageCollections/Storage/SQLitePackageCollectionsStorage.swift]: I'm ok with logging rather than failure.

**Pull request merged**